lib/nexpose/rexlite/mime/message.rb

```diff
@@ -0,0 +1,162 @@
+# -*- coding: binary -*-
+module Rexlite
+module MIME
+class Message
+
+  require 'nexpose/rexlite/mime/header'
+  require 'nexpose/rexlite/mime/part'
+  require 'nexpose/rexlite/mime/encoding'
+
+  include Rexlite::MIME::Encoding
+
+  attr_accessor :header, :parts, :bound, :content
+
+
+  def initialize(data=nil)
+    self.header = Rexlite::MIME::Header.new
+    self.parts  = []
+    self.bound  = "_Part_#{rand(1024)}_#{rand(0xffffffff)}_#{rand(0xffffffff)}"
+    self.content = ''
+    if data
+      head,body = data.split(/\r?\n\r?\n/, 2)
+
+      self.header.parse(head)
+      ctype = self.header.find('Content-Type')
+
+      if ctype && ctype[1] && ctype[1] =~ /multipart\/mixed;\s*boundary="?([A-Za-z0-9'\(\)\+\_,\-\.\/:=\?^\s]+)"?/
+        self.bound = $1
+        chunks = body.to_s.split(/--#{self.bound}(--)?\r?\n/)
+        self.content = chunks.shift.to_s.gsub(/\s+$/, '')
+        self.content << "\r\n" unless self.content.empty?
+
+        chunks.each do |chunk|
+          break if chunk == "--"
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.